conans/test/unittests/client/generators/json_test.py

```diff
@@ -17,12 +17,15 @@ def variables_setup_test(self):
         conanfile.initialize(Settings({}), EnvValues())
         ref = ConanFileReference.loads("MyPkg/0.1@lasote/stables")
         cpp_info = CppInfo("dummy_root_folder1")
+        cpp_info.name = ref.name  # 'name' value in CppInfo has to be assigned before
```

Comment:
This change in the test is needed due to the new feature of the user declaring a different name for a library ("target name"):
The output of this generator will change for the *name* variable in case a recipe sets a different name not corresponding to the same one in the reference.